### PR TITLE
support configure cpu.idle by Cgroupfs

### PR DESCRIPTION
**Pull request by wineway**

### CODE REVIEW

File: crates/libcontainer/Cargo.toml [hunk 1]
```diff
@@ -29,7 +29,7 @@ libc = "0.2.125"
 log = "0.4"
 mio = { version = "0.8.3", features = ["os-ext", "os-poll"] }
 nix = "0.24.1"
-oci-spec = { git = "https://github.com/containers/oci-spec-rs", rev = "89376af" }
+oci-spec = { git = "https://github.com/containers/oci-spec-rs", rev = "6df620e" }
```
Comment (utam0k) [inline crates/libcontainer/Cargo.toml:32]:
```suggestion
oci-spec = "0.5.6"
```

File: crates/youki/Cargo.toml [hunk 2]
```diff
@@ -25,7 +25,7 @@ libcontainer = { version = "0.0.3", path = "../libcontainer" }
 liboci-cli = { version = "0.0.3", path = "../liboci-cli" }
 log = { version = "0.4", features = ["std"]}
 nix = "0.24.1"
-oci-spec = { git = "https://github.com/containers/oci-spec-rs", rev = "89376af" }
+oci-spec = { git = "https://github.com/containers/oci-spec-rs", rev = "6df620e" }
```
Comment (utam0k) [inline crates/youki/Cargo.toml:28]:
```suggestion
oci-spec = "0.5.6"
```

File: crates/libcontainer/Cargo.toml [hunk 3]
```diff
@@ -43,7 +43,7 @@ wasmer = { version = "2.2.0", optional = true }
 wasmer-wasi = { version = "2.2.0", optional = true }
 
 [dev-dependencies]
-oci-spec = { git = "https://github.com/containers/oci-spec-rs", rev = "89376af", features = ["proptests"] }
+oci-spec = { git = "https://github.com/containers/oci-spec-rs", rev = "6df620e", features = ["proptests"] }
```
Comment (utam0k) [inline crates/libcontainer/Cargo.toml:46]:
```suggestion
oci-spec = { version = "0.5.6", features = ["proptests"] }
```

File: crates/libcgroups/Cargo.toml [hunk 4]
```diff
@@ -25,7 +25,7 @@ nix = "0.24.1"
 procfs = "0.12.0"
 log = "0.4"
 anyhow = "1.0"
-oci-spec = { git = "https://github.com/containers/oci-spec-rs", rev = "89376af" }
+oci-spec = { git = "https://github.com/containers/oci-spec-rs", rev = "6df620e" }
```
Comment (utam0k) [inline crates/libcgroups/Cargo.toml:28]:
```suggestion
oci-spec = "0.5.6"
```

File: crates/libcgroups/Cargo.toml [hunk 5]
```diff
@@ -35,7 +35,7 @@ errno = { version = "0.2.8", optional = true }
 libc = { version = "0.2.125", optional = true }
 
 [dev-dependencies]
-oci-spec = { git = "https://github.com/containers/oci-spec-rs", rev = "89376af", features = ["proptests"] }
+oci-spec = { git = "https://github.com/containers/oci-spec-rs", rev = "6df620e", features = ["proptests"] }
```
Comment (utam0k) [inline crates/libcgroups/Cargo.toml:38]:
```suggestion
oci-spec = { version = "0.5.6", features = ["proptests"] }
```

**Review by utam0k — COMMENTED** [review]
@wineway Thanks for your first contribution. I left some comments. If you are okay, please add some integration tests? If you don't have time to add it, please tell me.
https://github.com/containers/youki/tree/main/tests/rust-integration-tests/integration_test/src/tests/cgroups/cpu

File: crates/libcgroups/src/v1/cpu.rs [hunk 6]
```diff
@@ -40,6 +41,7 @@ impl Controller for Cpu {
                 || cpu.quota().is_some()
                 || cpu.realtime_period().is_some()
                 || cpu.realtime_runtime().is_some()
+                || cpu.idle().is_some()
```
Comment (tsturzl) [inline crates/libcgroups/src/v1/cpu.rs:44]: Should either add or adapt the test cases to account for setting and checking this new setting.

File: crates/libcgroups/src/v2/cpu.rs [hunk 7]
```diff
@@ -96,6 +97,10 @@ impl Cpu {
             common::write_cgroup_file(path.join(CGROUP_CPU_BURST), burst)?;
         }
 
+        if let Some(idle) = cpu.idle() {
+            common::write_cgroup_file(path.join(CGROUP_CPU_IDLE), idle)?;
```
Comment (tsturzl) [inline crates/libcgroups/src/v2/cpu.rs:101]: Same here. We should add or adapt the test cases to account for the new configuration.

**Review by tsturzl — CHANGES_REQUESTED** [review]

**Comment by wineway** [comment]
@utam0k @tsturzl updated! PTAL

**Comment by utam0k** [comment]
> and seems tag 0.5.6 doesn't contains https://github.com/containers/oci-spec-rs/pull/103, so I edited commit messa

Oops, sorry, I failed to include your PR when I released oci-spec-rs 😭

**Review by utam0k — APPROVED** [review]
💯

**Comment by wineway** [comment]
@utam0k Can you help trigger ci again? I don't have a cgroupv1 env, and I would like to ensure integration test is correct😂

**Comment by utam0k** [comment]
@tsturzl Can I ask you to review it again?

**Pull request merged**